revert to import azure keyvault certs instead of generating them (#5)
mspnp/aks-secure-baseline: 07-workload-prerequisites.md

# Workflow Prerequisites

The AKS Cluster has been enrolled in [GitOps management](./06-gitops), wrapping up the infrastructure focus of the [AKS secure Baseline reference implementation](./). Follow the steps below to import the TLS certificate that the Ingress Controller will serve for Application Gateway to connect to your web app.

## Steps

> ## Import the wildcard certificate for the Ingress Controller to Azure KeyVault

Reviewer: ```suggestion
## Import the wildcard certificate for the AKS Ingress Controller to Azure Key Vault
```

> :book: Contoso Bicycle procured a CA certificate, a standard one, to be used with the AKS Ingress Controller. This one is not EV, as it will not be user facing.

1. Obtain the Azure Key Vault details and give the current user permissions to import certificates.

   > :book: Finally the app team decides to use a wildcard certificate of `*.aks-ingress.contoso.com` for the ingress controller. They use Azure Key Vault to import and manage the lifecycle of this certificate.

   ```bash
   KEYVAULT_NAME=$(az deployment group show --resource-group rg-bu0001a0008 -n cluster-stamp --query properties.outputs.keyVaultName.value -o tsv)
   az keyvault set-policy --certificate-permissions import list get -n $KEYVAULT_NAME --upn $(az account show --query user.name -o tsv)
   ```

> 1. Import the Cluster Ingress Controller's Wildcard Certificate for `*.aks-ingress.contoso.com`.

Reviewer: ```suggestion
1. Import the AKS Ingress Controller's Wildcard Certificate for `*.aks-ingress.contoso.com`.
```

>    :warning: If you already have access to an appropriate certificate, or can procure one from your organization, consider using it from this very same step. For more information, please take a look at the [import certificate tutorial using Azure Key Vault](https://docs.microsoft.com/azure/key-vault/certificates/tutorial-import-certificate#import-a-certificate-to-key-vault).

Reviewer: ```suggestion
   :warning: If you already have access to an appropriate certificate, or can procure one from your organization, consider using it for this step. For more information, please take a look at the [import certificate tutorial using Azure Key Vault](https://docs.microsoft.com/azure/key-vault/certificates/tutorial-import-certificate#import-a-certificate-to-key-vault).
```
